src/Nethermind/Nethermind.Blockchain/Producers/BaseLoopBlockProducer.cs

```diff
@@ -31,7 +31,7 @@ public abstract class BaseLoopBlockProducer : BaseBlockProducer
         private Task _producerTask;
         private readonly CancellationTokenSource _loopCancellationTokenSource = new CancellationTokenSource();
         private readonly CancellationTokenSource _stepCancellationTokenSource = new CancellationTokenSource();
-        private bool _canProduce;
+        private bool _canProduce = true;
```

Comment:
I had some nasty forks with this true as default when syncing pre-existing networks

Reply:
can we have proper integration tests for that? it is all substituted in test

Reply:
also if switching it to false fixes the problem then it means it only makes it appear less often IMHO

Reply:
there is additional check later that checks whether the queue is empty

Reply:
maybe set it true if _blocktree is at genesis?

Reply:
_canProduce = _blockTree.Head.Number == 0?

Reply:
No its additional multi-thread race condition. Generally you are right this might not be main issue. Main issue is probably why it doesn't reorg properly later.